code/_onclick/hud/action.dm

```diff
@@ -31,14 +32,16 @@
 		Remove(owner)
 	owner = T
 	T.actions += src
+	if (button)
+		button.moved = 0
 	T.update_action_buttons()
 
 /datum/action/proc/Remove(mob/living/T)
 	if(button)
 		if(T.client)
 			T.client.screen -= button
-		qdel(button)
-		button = null
+		T.contents -= button
```

Comment:
Space worms are unticked. Xenos use stomach_contents to prevent barfing out its own organs.